src/config.cc

```diff
@@ -47,6 +59,61 @@ std::pair<bool, size_t> parse_uint(const char* str) {
 	return {false, 0};
 }
 
+using device_config_vector = std::vector<size_t>;
```

Comment:
This does not create a new unique type, meaning that the specialization of `default_parser` and `default_validator` for this type applies to **all** `std::vector<size_t>` which is not a good idea, as other `vector<size_t>` might have different requirements.

For this use-case it probably makes the most sense to provide a custom parser and validator lambda to `register_variable` instead of specializing the `default_parser` and `default_validator`, but that would of course also be possible when wrapping the vector in a custom unique type.